File: dotnet/csharp/src/InventoryCommand.cs
using System;

namespace EventSourcing
{
    public abstract class InventoryCommand
    {
        public class Create : InventoryCommand
        {
            public Create(Guid inventoryId, string name)
            {
                InventoryId = inventoryId;
                Name = name;
            }

            public Guid InventoryId { get; }

            public string Name { get; }
        }

        public class Stock : InventoryCommand
        {
            public Stock(int count) => Count = count;

            public int Count { get; }

        }

        public class Sell : InventoryCommand
        {
            public Sell(int count) => Count = count;

            public int Count { get; }
        }

        public class Activate : InventoryCommand
        {
            public Activate() { }
        }

        public class Deactivate : InventoryCommand
        {
            public Deactivate() { }
        }
    }
}

File: node/src/inventoryCommand.js
class Create {
    constructor(inventoryId, name) {
        this.inventoryId = inventoryId;
        this.name = name;
    }
}

class Stock {
    constructor(count) {
        this.count = count;
    }
}

class Sell {
    constructor(count) {
        this.count = count;
    }
}

class Deactivate { }

class Activate { }

module.exports = { Create, Stock, Sell, Deactivate, Activate };